web/htdocs/assets/js/codesearch_ui.js

```diff
@@ -45,10 +47,84 @@ function shorten(ref) {
   return ref;
 }
 
-var Match = Backbone.Model.extend({
+var MatchView = Backbone.View.extend({
+  tagName: 'div',
   initialize: function() {
+    this.model.on('change', this.render, this);
   },
-  url: function() {
+  render: function() {
+    var div = this._render();
+    this.setElement(div);
+    return this;
+  },
+  _renderLno: function(n, isMatch) {
+    return h.a({cls: 'lno', href: this.model.url(n)}, [n.toString(), isMatch ? ":" : "-"]);
+  },
+  _render: function() {
+    var i;
+    var ctx_before = [], ctx_after = [];
+    var lno = this.model.get('lno');
+    var ctxBefore = this.model.get('context_before'), clip_before = this.model.get('clip_before');
+    var ctxAfter = this.model.get('context_after'), clip_after = this.model.get('clip_after');
+
+    var lines_to_display_before = Math.max(0, ctxBefore.length - (clip_before || 0));
+    for (i = 0; i < lines_to_display_before; i ++) {
+      ctx_before.unshift(h.div([
+                                 this._renderLno(lno - i - 1, false),
+                                 this.model.get('context_before')[i]
+                               ]));
+    }
+    var lines_to_display_after = Math.max(0, ctxAfter.length - (clip_after || 0));
+    for (i = 0; i < lines_to_display_after; i ++) {
+      ctx_after.push(h.div([
+                             this._renderLno(lno + i + 1, false),
+                             this.model.get('context_after')[i]
+                           ]));
+    }
+    var line = this.model.get('line');
+    var bounds = this.model.get('bounds');
+    var pieces = [line.substring(0, bounds[0]),
+                  line.substring(bounds[0], bounds[1]),
+                  line.substring(bounds[1])];
+
+    var classes = ['match'];
+    if(clip_before !== undefined) classes.push('clip-before');
+    if(clip_after !== undefined) classes.push('clip-after');
+
+    var matchElement = h.div({cls: classes.join(' ')}, [
```

Comment:
Yikes.

My suggestion would be to merge the best version you have without worrying about IE, and file any known issues with it in the tracker. Then if someone comes along who has a better idea of how to make this work uniformly, they'll have all the information to take a crack at it.